src/sheet/wire-item.c

```diff
@@ -166,9 +166,6 @@ static void wire_item_dispose (GObject *object)
 
 	priv = WIRE_ITEM (object)->priv;
 
```

Comment:
What's the reason to remove this? It calls unref and sets the pointer to `NULL`

Reply:
It leads to a segmentation fault when a wire is deleted, as the three objects are already unreferenced by the parent class dispose function.

Not convinced yet ? Please refer to the above mentioned stack trace: https://github.com/drahnr/oregano/pull/219#discussion_r138711186

Reply:
The final result is the same as you intended (object is unreferenced), the only practical difference is that the pointer to the unreferenced object is not set to NULL. The latter is probably not essential in this case...

Reply:
See this patch as a whole and consider we now have the item unreferencing in sheet_delete_selection().

Reply:
Please squash the patches which create a proper changeset which compiles. It is hard to track a lot of small seemingly unrelated commits which actually shift functionality.

Reply:
It is not possible to squash commits that are far apart from each other...

However, for your information, the above mentioned commit:

https://github.com/gtrentalancia/oregano/commit/f3cfd251a348e9349880eb74d4b1b406f900cd7e

is related to the following two (far apart, cannot be squashed):

https://github.com/gtrentalancia/oregano/commit/ad8b1648e89f5ba4eb291f52c7b54ad7a66084c4

and

https://github.com/gtrentalancia/oregano/commit/2311e1818ba3742e7993fafd083060ca5c45abe1

in that it improves the original idea of the second one (to deallocate the item and not only the item-data).

I hope it helps....

Reply:
I have now squashed as required !

Please merge if possible... Thanks.

Reply:
In the newly rebased git tree, you want to look at the following commit:

https://github.com/gtrentalancia/oregano/commit/c65fdf9552ccfee154156c13facf8f654037141c